src/services/JobCandidateService.js

```diff
@@ -358,11 +360,50 @@ searchJobCandidates.schema = Joi.object().keys({
   }).required()
 }).required()
 
+/**
+ * Download jobCandidate resume
+ * @params {Object} currentUser the user who perform this operation
+ * @params {String} id the jobCandidate id
+ */
+async function downlaodJobCandidateResume (currentUser, id) {
+  const jobCandidate = await JobCandidate.findById(id)
+  const { id: currentUserUserId } = await helper.getUserByExternalId(currentUser.userId)
+
+  // customer role
+  if (!jobCandidate.viewedByCustomer && currentUserUserId !== jobCandidate.userId && currentUser.roles.length === 1 && currentUser.roles[0] === TopCoderUserPermissionRole) {
+    const job = await models.Job.findById(jobCandidate.jobId)
+    const { handle } = await helper.getUserById(jobCandidate.userId, true)
+    const { email } = await helper.getMemberDetailsByHandle(handle)
+
+    await EmailNotificationService.sendEmail(currentUser, {
+      template: 'taas.notification.job-candidate-resume-viewed',
+      recipients: [email],
+      data: {
+        jobCandidateUserHandle: handle,
+        jobName: job.title,
+        notificationType: {
+          jobCandidateResumeViewed: true
+        }
+      }
+    })
+
+    await updateJobCandidate(currentUser, jobCandidate.id, { viewedByCustomer: true })
+  }
```

Comment:
Let's wrap it in `try / catch` and if some error happened during email notification sending we just log error, but still error resume downloading.